---
title: "Consensus"
output: html_document
---

#chargement package
```{r}
library(cowplot)
library(xlsx)
library(tidyverse)
library(questionr)
library(survival)
library(survminer)
```



#chargement des données

```{r}
#fichier contenant les données donneur
meta_data <- read.xlsx("/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pcawg_donor_clinical_August2016_v9.xlsx", 1, header=TRUE)
meta_data <- meta_data[,!colnames(meta_data)==c("NA.", "NA..1")]
#fichier contenant id echantillon et id donneurs (et autres info relatives aux samples)
sample_sheet <- read.table("/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pcawg_sample_sheet.tsv", sep="\t", header=TRUE)
#fichier vcf avec la ligne qu'on veut
vcf_file <- "/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pol_1kb.vcf"
vcf_lines <- readLines(vcf_file)
histology <- read.xlsx("/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pcawg_specimen_histology_August2016_v9.xlsx", 1, header=TRUE)
histology <- histology[,!colnames(histology)==c("NA.", "NA..1")]


#boucle pour récupérer le nom des échantillons 
j=1
tab = NULL
info=NULL
for (i in 1:length(vcf_lines)){
    #ligne actuelle
    line <- vcf_lines[i]
    #pour les noms des échantillons
    if(str_detect(line,"^#{1}[^#]")){
      list = str_split(as.character(line), "\t")
      list_info = list[[1]][1:9]
      list = list[[1]][-(1:9)]
      
    }
    #on garde l'autre ligne (avec genotype etc)
    else if(!str_detect(line,"^#")){
      
      pol = str_split(as.character(line), "\t")
      j <- j+1
      #autres lignes 
      if(j>2){
        #info
        info <- cbind(info, pol[[1]][1:9])
        colnames(info)[j-1] <- info[2,j-1]
        
        #on garde que l'information du génotype
        tab <- cbind(tab, pol[[1]][10:length(pol[[1]])])
      }
      #création des tables 
      else{ 
        info <- pol[[1]][1:9]
        tab <- cbind(list, pol[[1]][10:length(pol[[1]])])
      }
    }
}

colnames(tab) <- c("Echantillon", info[2,])
colnames(info) <- info[2,]
rownames(info) <- list_info 

tab <- as.data.frame(tab)
info <- as.data.frame(info)


#on recode les génotypes hétérozygotes
tab[,2:length(tab)] <- sapply(tab[,2:ncol(tab)],function(x) fct_recode(x, "0|1" = "1|0"))


#fusions de stables ==> nom ech -> nom donneur -> genotype -> métadata
fusion <- left_join(tab, sample_sheet[,c("aliquot_id", "icgc_donor_id", "icgc_sample_id")], by=c(Echantillon="aliquot_id")) %>% left_join(meta_data)  %>% filter(donor_wgs_included_excluded=="Included")

histology <- histology %>% filter(specimen_library_strategy=="WGS")
fusion <- left_join(fusion, histology[,c(9,12:ncol(histology))])
fusion <- fusion[-(which(duplicated(fusion$Echantillon))), ]

#petites stats
table(is.na(fusion$donor_vital_status))
table(is.na(fusion$donor_survival_time))
length(intersect(which(is.na(fusion$donor_vital_status)), which(is.na(fusion$donor_survival_time))))
try <- setdiff(which(is.na(fusion$donor_vital_status)), which(is.na(fusion$donor_survival_time)))
fusion[try,]

str(fusion)
fusion[,2:ncol(tab)] <- lapply(2:ncol(tab), function(l) as.factor(fusion[,l]))
fusion$icgc_donor_id <- as.character(fusion$icgc_donor_id)
fusion$submitted_donor_id <- as.character(fusion$submitted_donor_id)
fusion$X..donor_unique_id <- as.character(fusion$X..donor_unique_id)
fusion$tcga_donor_uuid <- as.character(fusion$tcga_donor_uuid)
fusion$icgc_sample_id <- as.character(fusion$icgc_sample_id)
```

# quick overview & graphs
```{r}
#permet de faire différents plots sur les snps/indels 
#k est l'indice de la colonne et data le tableau
plots_snps = function(k, data){
            dvs <- ggplot(data) + geom_bar(aes(x = data[,k], fill = donor_vital_status)) + xlab(colnames(data)[k]) + ggtitle(paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
            dvs_fill <- ggplot(data) + geom_bar(aes(x = data[,k], fill = donor_vital_status), position="fill") + xlab(colnames(data)[k]) 
            plot_grid(dvs, dvs_fill, ncol = 2, nrow = 1)
            ggsave(paste(colnames(data)[k],"donor_vital_status.pdf", sep="_"), device="pdf", path="./plots", width=15)
            
            both <- ggplot(data) +  geom_bar(aes(x = data[,k], fill = donor_vital_status)) + facet_wrap(~project_code) + ggtitle(paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" ")) + theme(legend.position="bottom") + xlab(colnames(data)[k])
            both_fill <- ggplot(data) +  geom_bar(aes(x = data[,k], fill = donor_vital_status), position="fill") + facet_wrap(~project_code) + theme(legend.position="bottom") + xlab(colnames(data)[k])
            plot_grid(both, both_fill, ncol = 2, nrow = 1)
            ggsave(paste(colnames(data)[k],"donor_vital_status_project_code.pdf", sep="_"), device="pdf", path="./plots", width=15)
            
            bp <-  ggplot(data) + geom_bar(aes(x = project_code, fill = data[,k])) + ggtitle(paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
            bp_fill <- ggplot(data) + geom_bar(aes(x = project_code, fill = data[,k]), position="fill")
            plot_grid(bp, bp_fill, ncol = 1, nrow = 2)
            ggsave(paste(colnames(data)[k],"project_code.pdf", sep="_"), device="pdf", path="./plots", width=20)
}

#fonction qui analyse le tableau entièrement (et réutilise plot au dessus)
help = function(k, data){
  if(is.factor(data[,k])){
    
    #print(data[,k])
    write.table(freq(data[,k]), paste("./freq/freq_", colnames(data)[k],".txt"))

    ##pour représenter les différents snp/indel
    if(k>=2 && k<=ncol(tab)){ 
      ##on représente que les snp/indels qui ont geno >5% 
      if(nrow(freq(data[,k]))>1){
          if (!(nrow(freq(data[,k]))==2 && (freq(data[,k])$'%'[1]<5  || freq(data[,k])$'%'[2]<5))){ 
            plots_snps(k, data)
         }
      }
     
    }
    else {
      ggplot(data) + geom_bar(aes(x = data[,k], fill = donor_vital_status)) + xlab(colnames(data)[k]) + theme(legend.position="bottom") + ggtitle(paste(colnames(data)[k], sep=" "))
      ggsave(paste(colnames(data)[k], "_vital_status.pdf", sep=""), device="pdf", path="./plots", width=15)

    }
  }
  else if(is.numeric(data[,k])){
    print(pdf(paste("./plots/",colnames(data)[k],".pdf", sep="")))
    hist(data[,k], main=paste(colnames(data)[k], sep=" "), xlab=colnames(data)[k])
    dev.off()
  }
}

#system.time(
 sapply(seq_len(ncol(fusion)), help, data=fusion)
#) 
#37.046 sec

#system.time(
  #for (i in 1:ncol(fusion)){
    #help(i, fusion)
  #}
#)
#38.454 sec

```

#Fisher
```{r}
#?fisher.test

#variable = le nom de la colonne qu'on veut regarder
fisher <- function(variable){
  mat_pvalue <- matrix(NA, ncol = ncol(tab)-1, nrow = 2)
  colnames(mat_pvalue) <- colnames(tab[2:ncol(tab)])
  rownames(mat_pvalue)<- c(paste("pvalue", variable, sep="_"), paste("pvalue_adjust", variable, sep="_"))

    for (k in 2:ncol(tab)){
      print(colnames(fusion)[k])
      #on enlève là où il y a un seul génotype
      if(nrow(freq(fusion[,k]))>1 ){
        conting <- table(fusion[,k], fusion[,which(colnames(fusion)==variable)])
        #write.table(conting, paste("./Fisher/snp_vs_cohorte/conting/", colnames(fusion)[k],".txt", sep=""))

        conting_freq <- prop.table(table(fusion[,k], fusion[,which(colnames(fusion)==variable)]), margin=2)
        #write.table(conting_freq, paste("./Fisher/snp_vs_cohorte/conting_freq/", colnames(fusion)[k],".txt", sep=""))

        test <- try(fisher.test(conting), silent = TRUE)
        if (is(test) == "try-error") {
          print(" Direct p-value computation failed. Trying in simulated mode (MCMC).")
          test <- fisher.test(conting, simulate.p.value=T, B=1E+6)
        }
        mat_pvalue[1,k-1] <- test$p.value 
      }
      else{
        print("1 seul génotype")
      }
    }  
  mat_pvalue[2, ]  <- p.adjust(mat_pvalue[1,], method = "BH")
  return(mat_pvalue)
}

mat_pvalue_cohorte = NULL
mat_pvalue_cohorte <- fisher("project_code")
#write.table(mat_pvalue_cohorte, "./mat_pvalue_snp_vs_cohorte.txt")

signif_cohorte <- colnames(mat_pvalue_cohorte)[which(mat_pvalue_cohorte[1,] < 0.05)]
signif_cohorte_adjust <- colnames(mat_pvalue_cohorte)[which(mat_pvalue_cohorte[2,] < 0.05)]
length(signif_cohorte_adjust)


```

## Fisher 1 maladie vs snp avec inclusion de 10 snps randoms 
```{r}

random_file <- "/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/random.vcf"
random_lines <- readLines(random_file)

#boucle pour récupérer le nom des échantillons 
j=1
random = NULL
info_random=NULL
pol_random = NULL
list=NULL
for (i in 1:length(random_lines)){
    #ligne actuelle
    line <- random_lines[i]
    #on garde l'autre ligne (avec genotype etc)
    if(!str_detect(line,"^#")){
      
      pol_random = str_split(as.character(line), "\t")
      j <- j+1
      #autres lignes 
      if(j>2){
        #info
        info_random <- cbind(info_random, pol_random[[1]][1:9])
        colnames(info_random)[j-1] <- info_random[2,j-1]
        
        #on garde que l'information du génotype
        random <- cbind(random, pol_random[[1]][10:length(pol_random[[1]])])
      }
      #création des tables 
      else{ 
        info_random <- pol_random[[1]][1:9]
        random <- cbind(list, pol_random[[1]][10:length(pol_random[[1]])])
      }
    }
}

colnames(random) <-  info_random[2,]
colnames(info_random) <- info_random[2,]
rownames(info_random) <- list_info 

random <- as.data.frame(random)
info_random <- as.data.frame(info_random)

tab_random <- cbind(tab, random)

#fusions de stables ==> nom ech -> nom donneur -> genotype -> métadata
fusion_random <- left_join(tab_random, sample_sheet[,c("aliquot_id", "icgc_donor_id", "icgc_sample_id")], by=c(Echantillon="aliquot_id")) %>% left_join(meta_data)  %>% filter(donor_wgs_included_excluded=="Included")

fusion_random[,2:length(tab_random)] <- sapply(fusion_random[,2:ncol(tab_random)],function(x) fct_recode(x, "0|1" = "1|0"))

fusion_random <- left_join(fusion_random, histology[,c(9,12:ncol(histology))])
fusion_random <- fusion_random[-(which(duplicated(fusion_random$Echantillon))), ]



names=NULL
pvalues=NULL

for (i in levels(fusion_random$project_code)){
  pvalue=NULL
  lignes = which(fusion_random$project_code==i)
  print(i)
  
  if(length(lignes)>0){
    names <- c(names, i)
    conting = NULL
    conting1 = NULL
    conting2 =NULL
    conting3 =NULL
    conting_random = NULL
    for (j in 2:length(tab)){
      conting <- cbind(conting, table(factor(fusion_random[lignes,j], levels=c("0|0", "0|1", "1|1"))))
    }
    for(k in which(colnames(tab_random) %in% colnames(random))){
      conting_random <- cbind(conting_random, table(factor(fusion_random[lignes,k], levels=c("0|0", "0|1", "1|1"))))
    }
    conting <- cbind(conting, rowSums(conting_random))
    colnames(conting) = c(colnames(tab_random)[2:length(tab)], "random")
    write.table(conting, paste("./Fisher/cohorte_vs_snp/conting/",i,".txt", sep=""))
    
    #comme c'est trop long en normal ou en hybride on fait directement une simulation
    test <- try(fisher.test(as.matrix(conting), hybrid=TRUE, simulate.p.value=T, B=1E+6), silent = TRUE)
    
    #if (is(test) == "try-error") {
     # print(" Direct p-value computation failed. Trying in simulated mode (MCMC).")
      #test <- fisher.test(as.matrix(conting), simulate.p.value=T, B=1E+6)
    #}
    pvalues <- c(pvalues,test$p.value)
  }
}
mat_pvalue <- matrix(c(pvalues, p.adjust(pvalues, method = "BH")), nrow=2, ncol=length(pvalues), byrow=TRUE)
rownames(mat_pvalue) <- c("pvalue", "pvalue_adjust")
colnames(mat_pvalue) <- names

write.table(mat_pvalue, "./Fisher/mat_pvalue_cohorte_vs_snp.txt")

```





#prepare data
```{r}
fusion_na <- fusion[!is.na(fusion$donor_survival_time),]
fusion_na$death = ifelse(fusion_na$donor_vital_status=="alive", 0, 1)
#fusion_na$death <- as.factor(fusion_na$death)
```


#KM

```{r message=FALSE, warning=FALSE, paged.print=FALSE}

#survie globale
KM<-survfit(Surv(donor_survival_time, death) ~ 1,data=fusion_na, conf.type="log-log")
survpd <- ggsurvplot(KM,risk.table = TRUE,break.time.by = 1000)
ggsave(plot=print(survpd), "Survie_globale_days.pdf", device="pdf", path="./survie", width=15)

KMy<-survfit(Surv(donor_survival_time/365.25, death) ~ 1,data=fusion_na, conf.type="log-log")
survpy <- ggsurvplot(KMy,risk.table = TRUE)
ggsave(plot=print(survpy), "Survie_globale_years.pdf", device="pdf", path="./survie", width=15)


#selon les projets
KM_project<-survfit(Surv(donor_survival_time, death) ~ project_code, data=fusion_na, conf.type="log-log")
survpd <- ggsurvplot(KM_project,risk.table = TRUE,break.time.by = 1000)
ggsave(plot=print(survpd), "Survie_cohorte_days.pdf", device="pdf", path="./survie", width=15)


KMy_project<-survfit(Surv(donor_survival_time/365.25, death) ~ project_code ,data=fusion_na, conf.type="log-log")
survpy <- ggsurvplot(KMy_project,risk.table = TRUE)
ggsave(plot=print(survpy), "Survie_cohorte_years.pdf", device="pdf", path="./survie", width=15)

  
survie <- function(k, data){
  #survie cohorte
  #en jours
  KM_geno<- survfit(Surv(donor_survival_time, death) ~ data[,k], data= data, conf.type="log-log")
  survpd <- ggsurvplot(KM_geno,risk.table = TRUE,break.time.by = 1000, data=data, title=paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
  ggsave(plot=print(survpd), paste(colnames(data)[k], "days_geno.pdf", sep="_"), device="pdf", path="./survie", width=15)

  #en années
  KMy_geno<-survfit(Surv(donor_survival_time/365.25, death) ~ data[,k], data=data, conf.type="log-log")
  survpy <- ggsurvplot(KMy_geno,risk.table = TRUE,  data=data, title=paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" ")) 
  ggsave(plot=print(survpy), paste(colnames(data)[k], "years_geno.pdf", sep="_"), device="pdf", path="./survie", width=15)

  #survie genotype + cohorte
  #KM_geno_project<-survfit(Surv(donor_survival_time, death) ~ data[,k] + project_code, data=data, conf.type="log-log")
  #survpd <- ggsurvplot(KM_geno_project,risk.table = TRUE,break.time.by = 1000,  data=data, title= paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
  #ggsave(plot=print(survpd), paste(colnames(data)[k], "days_geno_cohorte.pdf", sep="_"), device="pdf", path="./survie", width=15)

  #KMy_geno_project<-survfit(Surv(donor_survival_time/365.25, death) ~ data[,k] + project_code , data=data, conf.type="log-log")
  #survpy <- ggsurvplot(KMy_geno_project,risk.table = TRUE,  data=data, title=paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
  #ggsave(plot=print(survpy), paste(colnames(data)[k], "years_geno_cohorte.pdf", sep="_"), device="pdf", path="./survie", width=15)

}

sapply(2:ncol(tab), survie, data= fusion_na)


#survie par cohorte
signif=NULL
signif <- signif_cohorte_adjust[-c(1,3)] #après analyse deux snps ne montrent rien de particulier

#fusion_na[,2:ncol(tab)] <- lapply(2:ncol(tab), function(l) as.numeric(fusion_na[,l]))

mat_pvalue_survie <- matrix(NA, ncol = length(signif), nrow = length(names), dimnames=list(names, signif))
mat_pvalue_survie1 <- matrix(NA, ncol = length(signif), nrow = length(names), dimnames=list(names, signif))
mat_pvalue_survie2 <- matrix(NA, ncol = length(signif), nrow = length(names), dimnames=list(names, signif))
mat_pvalue_survie3 <- matrix(NA, ncol = length(signif), nrow = length(names), dimnames=list(names, signif))

for (i in names){
  print(i)
  lignes = which(fusion_na$project_code==i)
  
  if (length(lignes) > 0){
    if(!dir.exists(paste("./Survie/Nouveau/", i, sep=""))){
      dir.create(paste("./Survie/Nouveau/", i, sep=""))
    }
    else{
      unlink(paste("./Survie/Nouveau/", i, sep=""), recursive=TRUE)
      dir.create(paste("./Survie/Nouveau/", i, sep=""))
    }
    for (j in signif){
      print(j)
      #3 génotypes (si il y a)
      data <- fusion_na[lignes, ]
      #0|0 vs 0|1
      data1 <- data[which(data[,j] %in% c("0|0", "0|1")),]
      #0|0 vs 1|1
      data2 <- data[which(data[,j] %in% c("0|0", "1|1")),]
      #0|1 vs 1|1 
      data3 <- data[which(data[,j] %in% c("0|1", "1|1")),]
      
      # #en jours
      # KM_geno<- survfit(Surv(donor_survival_time, death) ~  data[,j], data= data, conf.type="log-log")
      # KM_geno1<- survfit(Surv(donor_survival_time, death) ~  data[,j], data= data1, conf.type="log-log")
      # KM_geno2<- survfit(Surv(donor_survival_time, death) ~  data[,j], data= data2, conf.type="log-log")
      # KM_geno3<- survfit(Surv(donor_survival_time, death) ~  data[,j], data= data3, conf.type="log-log")
      

      # survpd <- try(ggsurvplot(KM_geno,risk.table = TRUE, break.time.by = 1000, data=data, conf.int = TRUE, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")), silent=TRUE)
      # if(!is(survpd)=="try-error"){
      #   try <- try(ggsave(plot=print(survpd), paste(i,j, "days_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/nouveau/",i,"/", sep=""), width=15), silent=TRUE)
      # 
      #   if (is(try) == "try-error") {
      #     print("Une seule donnée, impossible de faire un interval de confiance")
      #     survpd <- ggsurvplot(KM_geno,risk.table = TRUE, break.time.by = 1000, data=data, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" "))
      # 
      #     ggsave(plot=print(survpd), paste(i,j, "days_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/nouveau/",i,"/", sep=""), width=15)
      #   }
      # 
      # }
        
      
      #en années
      #data
      KMy_geno<-survfit(Surv(donor_survival_time/365.25, death) ~ data[,j], data=data, conf.type="log-log")
      mat_pvalue_survie[i,j] <- try(surv_pvalue(KMy_geno, data=data)$pval, silent=TRUE)
      survpy <- try(ggsurvplot(KMy_geno,risk.table = TRUE,  data=data,conf.int = TRUE, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")), silent=TRUE)
      
      fitCox0 <-coxph(Surv(donor_survival_time, death) ~ data[,j], data=data)
      save(fitCox0, file=paste("./Survie/nouveau/",i,"/cox_", i,"_", j, ".RData", sep=""))
      
      if(!is(survpy)=="try-error"){
        try <- try(ggsave(plot=print(survpy), paste(i, j, "years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15), silent=TRUE)
      
        if (is(try) == "try-error") {
          print("Une seule donnée, impossible de faire un interval de confiance")
          survpy <- ggsurvplot(KMy_geno,risk.table = TRUE,  data=data, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")) 
          ggsave(plot=print(survpy), paste(i, j, "years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15)
        }
        
      }
      else{
        print("Impossible de représenter" )
      }
      
      
      #data1
      if(length(unique(data1[,j]))>1 && !all(unique(data[,j]) %in% unique(data1[,j]))){
        KMy_geno<-survfit(Surv(donor_survival_time/365.25, death) ~ data1[,j], data=data1, conf.type="log-log")
        mat_pvalue_survie1[i,j] <- try(surv_pvalue(KMy_geno, data=data1)$pval, silent=TRUE)
        survpy <- try(ggsurvplot(KMy_geno,risk.table = TRUE,  data=data1,conf.int = TRUE, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")), silent=TRUE)
        
        fitCox01 <-coxph(Surv(donor_survival_time, death) ~ data[,j], data=data)
        save(fitCox01, file=paste("./Survie/nouveau/",i,"/cox_0|0_vs_0|1", i,"_", j, ".RData", sep=""))
        
        if(!is(survpy)=="try-error"){
          try <- try(ggsave(plot=print(survpy), paste(i, j, "0|0_vs_0|1_years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15), silent=TRUE)
        
          if (is(try) == "try-error") {
            print("Une seule donnée, impossible de faire un interval de confiance")
            survpy <- ggsurvplot(KMy_geno,risk.table = TRUE,  data=data1, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")) 
            ggsave(plot=print(survpy), paste(i, j, "0|0_vs_0|1_years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15)
          }
          
        }
        else{
          print("Impossible de représenter" )
        }
      }
      
      
      #data2
      if(length(unique(data2[,j]))>1 && !all(unique(data[,j]) %in% unique(data2[,j]))){
        KMy_geno<-survfit(Surv(donor_survival_time/365.25, death) ~ data2[,j], data=data2, conf.type="log-log")
        mat_pvalue_survie2[i,j] <- try(surv_pvalue(KMy_geno, data=data2)$pval, silent=TRUE)
        survpy <- try(ggsurvplot(KMy_geno,risk.table = TRUE,  data=data2,conf.int = TRUE, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")), silent=TRUE)
        
        fitCox02 <-coxph(Surv(donor_survival_time, death) ~ data[,j], data=data)
        save(fitCox02, file=paste("./Survie/nouveau/",i,"/cox_0|0_vs_1|1", i,"_", j, ".RData", sep=""))
      
        if(!is(survpy)=="try-error"){
          try <- try(ggsave(plot=print(survpy), paste(i, j, "0|0_vs_1|1_years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15), silent=TRUE)
        
          if (is(try) == "try-error") {
            print("Une seule donnée, impossible de faire un interval de confiance")
            survpy <- ggsurvplot(KMy_geno,risk.table = TRUE,  data=data2, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")) 
            ggsave(plot=print(survpy), paste(i, j, "0|0_vs_1|1_years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15)
          }
          
        }
        else{
          print("Impossible de représenter" )
        }
      }
      
      #data3
      if(length(unique(data3[,j]))>1 && !all(unique(data[,j]) %in% unique(data3[,j]))){
        
        KMy_geno<-survfit(Surv(donor_survival_time/365.25, death) ~ data3[,j], data=data3, conf.type="log-log")
        mat_pvalue_survie3[i,j] <- try(surv_pvalue(KMy_geno, data=data3)$pval, silent=TRUE)
        survpy <- try(ggsurvplot(KMy_geno,risk.table = TRUE,  data=data3,conf.int = TRUE, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")), silent=TRUE)
        
        fitCox03 <-coxph(Surv(donor_survival_time, death) ~ data[,j], data=data)
        save(fitCox03, file=paste("./Survie/nouveau/",i,"/cox_0|1_vs_1|1", i,"_", j, ".RData", sep=""))
      
        if(!is(survpy)=="try-error"){
          try <- try(ggsave(plot=print(survpy), paste(i, j, "0|1_vs_1|1_years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15), silent=TRUE)
        
          if (is(try) == "try-error") {
            print("Une seule donnée, impossible de faire un interval de confiance")
            survpy <- ggsurvplot(KMy_geno,risk.table = TRUE,  data=data3, pval=TRUE, title=paste(i, j, info["REF", j], "-->", info["ALT", j], sep=" ")) 
            ggsave(plot=print(survpy), paste(i, j, "0|1_vs_1|1_years_geno.pdf", sep="_"), device="pdf", path=paste("./Survie/Nouveau/",i,"/", sep=""), width=15)
          }
          
        }
        else{
          print("Impossible de représenter" )
        }
      }
      
    }
  }
  else{
    print("Aucune donnée de survie")
  }
}

write.table(mat_pvalue_survie, "./Survie//Nouveau/mat_pvalue_all.txt")
write.table(mat_pvalue_survie1, "./Survie//Nouveau/mat_pvalue_0|0_vs_0|1.txt")
write.table(mat_pvalue_survie2, "./Survie//Nouveau/mat_pvalue_0|0_vs_1|1.txt")
write.table(mat_pvalue_survie3, "./Survie//Nouveau/mat_pvalue_0|1_vs_1|1.txt")

mat_pvalue_survie_ajdust <- matrix(p.adjust(mat_pvalue_survie, method="BH"), ncol = length(signif), nrow = length(names), dimnames=list(names, signif))
mat_pvalue_survie1_adjust <- matrix(p.adjust(mat_pvalue_survie1, method="BH"), ncol = length(signif), nrow = length(names), dimnames=list(names, signif))
mat_pvalue_survie2_ajdust <- matrix(p.adjust(mat_pvalue_survie2, method="BH"), ncol = length(signif), nrow = length(names), dimnames=list(names, signif))
mat_pvalue_survie3_ajdust <- matrix(p.adjust(mat_pvalue_survie3, method="BH"), ncol = length(signif), nrow = length(names), dimnames=list(names, signif))


write.table(mat_pvalue_survie_ajdust, "./Survie//Nouveau/mat_pvalue_ajust_all.txt")
write.table(mat_pvalue_survie1_adjust, "./Survie//Nouveau/mat_pvalue_ajust_0|0_vs_0|1.txt")
write.table(mat_pvalue_survie2_ajdust, "./Survie//Nouveau/mat_pvalue_ajust_0|0_vs_1|1.txt")
write.table(mat_pvalue_survie3_ajdust, "./Survie//Nouveau/mat_pvalue_ajust_0|1_vs_1|1.txt")

```

