---
title: "Consensus"
output: html_document
---

#chargement package
```{r}
library(cowplot)
library(xlsx)
library(tidyverse)
library(questionr)
library(survival)
library(survminer)
```



#chargement des données

```{r}
#fichier contenant les données donneur
meta_data <- read.xlsx("/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pcawg_donor_clinical_August2016_v9.xlsx", 1, header=TRUE)
meta_data <- meta_data[,!colnames(meta_data)==c("NA.", "NA..1")]
#fichier contenant id echantillon et id donneurs (et autres info relatives aux samples)
sample_sheet <- read.table("/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pcawg_sample_sheet.tsv", sep="\t", header=TRUE)
#fichier vcf avec la ligne qu'on veut
vcf_file <- "/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pol_1kb.vcf"
vcf_lines <- readLines(vcf_file)
histology <- read.xlsx("/Users/m_michel/Documents/stageM2/polymorphisme/ICGC/pcawg_specimen_histology_August2016_v9.xlsx", 1, header=TRUE)
histology <- histology[,!colnames(histology)==c("NA.", "NA..1")]


#boucle pour récupérer le nom des échantillons 
j=1
tab = NULL
info=NULL
for (i in 1:length(vcf_lines)){
    #ligne actuelle
    line <- vcf_lines[i]
    #pour les noms des échantillons
    if(str_detect(line,"^#{1}[^#]")){
      list = str_split(as.character(line), "\t")
      list_info = list[[1]][1:9]
      list = list[[1]][-(1:9)]
      
    }
    #on garde l'autre ligne (avec genotype etc)
    else if(!str_detect(line,"^#")){
      
      pol = str_split(as.character(line), "\t")
      j <- j+1
      #autres lignes 
      if(j>2){
        #info
        info <- cbind(info, pol[[1]][1:9])
        colnames(info)[j-1] <- info[2,j-1]
        
        #on garde que l'information du génotype
        tab <- cbind(tab, pol[[1]][10:length(pol[[1]])])
      }
      #création des tables 
      else{ 
        info <- pol[[1]][1:9]
        tab <- cbind(list, pol[[1]][10:length(pol[[1]])])
      }
    }
}

colnames(tab) <- c("Echantillon", info[2,])
colnames(info) <- info[2,]
rownames(info) <- list_info 

tab <- as.data.frame(tab)
info <- as.data.frame(info)


#on recode les génotypes hétérozygotes
tab[,2:length(tab)] <- sapply(tab[,2:ncol(tab)],function(x) fct_recode(x, "0|1" = "1|0"))


#fusions de stables ==> nom ech -> nom donneur -> genotype -> métadata
fusion <- left_join(tab, sample_sheet[,c("aliquot_id", "icgc_donor_id", "icgc_sample_id")], by=c(Echantillon="aliquot_id")) %>% left_join(meta_data)  %>% filter(donor_wgs_included_excluded=="Included")

histology <- histology %>% filter(specimen_library_strategy=="WGS")
fusion <- left_join(fusion, histology[,c(9,12:ncol(histology))])
fusion <- fusion[-(which(duplicated(fusion$Echantillon))), ]

#petites stats
table(is.na(fusion$donor_vital_status))
table(is.na(fusion$donor_survival_time))
length(intersect(which(is.na(fusion$donor_vital_status)), which(is.na(fusion$donor_survival_time))))
try <- setdiff(which(is.na(fusion$donor_vital_status)), which(is.na(fusion$donor_survival_time)))
fusion[try,]

str(fusion)
fusion[,2:ncol(tab)] <- lapply(2:ncol(tab), function(l) as.factor(fusion[,l]))
fusion$icgc_donor_id <- as.character(fusion$icgc_donor_id)
fusion$submitted_donor_id <- as.character(fusion$submitted_donor_id)
fusion$X..donor_unique_id <- as.character(fusion$X..donor_unique_id)
fusion$tcga_donor_uuid <- as.character(fusion$tcga_donor_uuid)
fusion$icgc_sample_id <- as.character(fusion$icgc_sample_id)
```

# quick overview & graphs
```{r}
#permet de faire différents plots sur les snps/indels 
#k est l'indice de la colonne et data le tableau
plots_snps = function(k, data){
            dvs <- ggplot(data) + geom_bar(aes(x = data[,k], fill = donor_vital_status)) + xlab(colnames(data)[k]) + ggtitle(paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
            dvs_fill <- ggplot(data) + geom_bar(aes(x = data[,k], fill = donor_vital_status), position="fill") + xlab(colnames(data)[k]) 
            plot_grid(dvs, dvs_fill, ncol = 2, nrow = 1)
            ggsave(paste(colnames(data)[k],"donor_vital_status.pdf", sep="_"), device="pdf", path="./plots", width=15)
            
            both <- ggplot(data) +  geom_bar(aes(x = data[,k], fill = donor_vital_status)) + facet_wrap(~project_code) + ggtitle(paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" ")) + theme(legend.position="bottom") + xlab(colnames(data)[k])
            both_fill <- ggplot(data) +  geom_bar(aes(x = data[,k], fill = donor_vital_status), position="fill") + facet_wrap(~project_code) + theme(legend.position="bottom") + xlab(colnames(data)[k])
            plot_grid(both, both_fill, ncol = 2, nrow = 1)
            ggsave(paste(colnames(data)[k],"donor_vital_status_project_code.pdf", sep="_"), device="pdf", path="./plots", width=15)
            
            bp <-  ggplot(data) + geom_bar(aes(x = project_code, fill = data[,k])) + ggtitle(paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
            bp_fill <- ggplot(data) + geom_bar(aes(x = project_code, fill = data[,k]), position="fill")
            plot_grid(bp, bp_fill, ncol = 1, nrow = 2)
            ggsave(paste(colnames(data)[k],"project_code.pdf", sep="_"), device="pdf", path="./plots", width=20)
}

#fonction qui analyse le tableau entièrement (et réutilise plot au dessus)
help = function(k, data){
  if(is.factor(data[,k])){
    
    #print(data[,k])
    write.table(freq(data[,k]), paste("./freq/freq_", colnames(data)[k],".txt"))

    ##pour représenter les différents snp/indel
    if(k>=2 && k<=ncol(tab)){ 
      ##on représente que les snp/indels qui ont geno >5% 
      if(nrow(freq(data[,k]))>1){
          if (!(nrow(freq(data[,k]))==2 && (freq(data[,k])$'%'[1]<5  || freq(data[,k])$'%'[2]<5))){ 
            plots_snps(k, data)
         }
      }
     
    }
    else {
      ggplot(data) + geom_bar(aes(x = data[,k], fill = donor_vital_status)) + xlab(colnames(data)[k]) + theme(legend.position="bottom") + ggtitle(paste(colnames(data)[k], sep=" "))
      ggsave(paste(colnames(data)[k], "_vital_status.pdf", sep=""), device="pdf", path="./plots", width=15)

    }
  }
  else if(is.numeric(data[,k])){
    print(pdf(paste("./plots/",colnames(data)[k],".pdf", sep="")))
    hist(data[,k], main=paste(colnames(data)[k], sep=" "), xlab=colnames(data)[k])
    dev.off()
  }
}

#system.time(
 sapply(seq_len(ncol(fusion)), help, data=fusion)
#) 
#37.046 sec

#system.time(
  #for (i in 1:ncol(fusion)){
    #help(i, fusion)
  #}
#)
#38.454 sec

```

#Fisher
```{r}
#?fisher.test

#variable = le nom de la colonne qu'on veut regarder
fisher <- function(variable){
  mat_pvalue <- matrix(NA, ncol = ncol(tab)-1, nrow = 1)
  colnames(mat_pvalue) <- colnames(tab[2:ncol(tab)])
  rownames(mat_pvalue)<- paste("pvalue", variable, sep="_")
  error_cohorte = NULL 

    for (k in 2:ncol(tab)){
      print(colnames(fusion)[k])
      #on enlève là où il y a un seul génotype
      if(nrow(freq(fusion[,k]))>1 ){
        conting <- table(fusion[,k], fusion[,which(colnames(fusion)==variable)])
        #write.table(conting, paste("./conting/", colnames(fusion)[k],".txt", sep=""))
        test <- try(fisher.test(conting), silent = TRUE)
        if (is(test) == "try-error") {
          print(" Direct p-value computation failed. Trying in simulated mode (MCMC).")
          test <- fisher.test(conting, simulate.p.value=T, B=1E+6)
          error <- c(error, colnames(fusion)[k])
        }
        mat_pvalue[1,k-1] <- test$p.value 
      }
      else{
        print("1 seul génotype")
      }
    }  
  return(mat_pvalue)
}

mat_pvalue_cohorte = NULL
mat_pvalue_cohorte <- fisher("project_code")
mat_pvalue_cohorte_adjust <- p.adjust(mat_pvalue_cohorte, method = "BH")

signif_cohorte <- colnames(mat_pvalue_cohorte)[which(mat_pvalue_cohorte < 0.05)]
signif_cohorte_adjust <- colnames(mat_pvalue_cohorte_adjust)[which(mat_pvalue_cohorte_adjust < 0.05)]
length(signif_cohorte_adjust)

mat_pvalue_death = NULL
mat_pvalue_death <- fisher("donor_vital_status")
mat_pvalue_death_adjust <- p.adjust(mat_pvalue_death, method = "BH")
signif_death <- colnames(mat_pvalue_death)[which(mat_pvalue_death < 0.05)]
signif_cohorte_adjust <- colnames(mat_pvalue_cohorte_adjust)[which(mat_pvalue_cohorte_adjust < 0.05)]


length(intersect(signif_cohorte, signif_death))
```


#prepare data
```{r}
fusion_na <- fusion[!is.na(fusion$donor_survival_time),]
fusion_na$death = ifelse(fusion_na$donor_vital_status=="alive", 0, 1)
```


#KM

```{r message=FALSE, warning=FALSE, paged.print=FALSE}

#survie globale
KM<-survfit(Surv(donor_survival_time, death) ~ 1,data=fusion_na, conf.type="log-log")
survpd <- ggsurvplot(KM,risk.table = TRUE,break.time.by = 1000)
ggsave(plot=print(survpd), "Survie_globale_days.pdf", device="pdf", path="./survie", width=15)

KMy<-survfit(Surv(donor_survival_time/365.25, death) ~ 1,data=fusion_na, conf.type="log-log")
survpy <- ggsurvplot(KMy,risk.table = TRUE)
ggsave(plot=print(survpy), "Survie_globale_years.pdf", device="pdf", path="./survie", width=15)


#selon les projets
KM_project<-survfit(Surv(donor_survival_time, death) ~ project_code, data=fusion_na, conf.type="log-log")
survpd <- ggsurvplot(KM_project,risk.table = TRUE,break.time.by = 1000)
ggsave(plot=print(survpd), "Survie_cohorte_days.pdf", device="pdf", path="./survie", width=15)


KMy_project<-survfit(Surv(donor_survival_time/365.25, death) ~ project_code ,data=fusion_na, conf.type="log-log")
survpy <- ggsurvplot(KMy_project,risk.table = TRUE)
ggsave(plot=print(survpy), "Survie_cohorte_years.pdf", device="pdf", path="./survie", width=15)

  
survie <- function(k, data){
  #survie cohorte
  KM_geno<- survfit(Surv(donor_survival_time, death) ~ data[,k], data= data, conf.type="log-log")
  survpd <- ggsurvplot(KM_geno,risk.table = TRUE,break.time.by = 1000, data=data, title=paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
  ggsave(plot=print(survpd), paste(colnames(data)[k], "days_geno.pdf", sep="_"), device="pdf", path="./survie", width=15)

  
  KMy_geno<-survfit(Surv(donor_survival_time/365.25, death) ~ data[,k], data=data, conf.type="log-log")
  survpy <- ggsurvplot(KMy_geno,risk.table = TRUE,  data=data, title=paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" ")) 
  ggsave(plot=print(survpy), paste(colnames(data)[k], "years_geno.pdf", sep="_"), device="pdf", path="./survie", width=15)

  #survie genotype + cohorte
  KM_geno_project<-survfit(Surv(donor_survival_time, death) ~ data[,k] + project_code, data=data, conf.type="log-log")
  survpd <- ggsurvplot(KM_geno_project,risk.table = TRUE,break.time.by = 1000,  data=data, title= paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
  ggsave(plot=print(survpd), paste(colnames(data)[k], "days_geno_cohorte.pdf", sep="_"), device="pdf", path="./survie", width=15)

  KMy_geno_project<-survfit(Surv(donor_survival_time/365.25, death) ~ data[,k] + project_code , data=data, conf.type="log-log")
  survpy <- ggsurvplot(KMy_geno_project,risk.table = TRUE,  data=data, title=paste(colnames(data)[k], info["REF", k-1], "-->", info["ALT", k-1], sep=" "))
  ggsave(plot=print(survpy), paste(colnames(data)[k], "years_geno_cohorte.pdf", sep="_"), device="pdf", path="./survie", width=15)

}

sapply(2:ncol(tab), survie, data= fusion_na)

```


#Cox
```{r}
cox <- function(k, data){
  if(nrow(freq(data[,k]))>1){
  
    fitCox0 <-coxph(Surv(donor_survival_time, death) ~ data[,k], data=data)
    
    fitCoxComplet <- coxph(Surv(donor_survival_time, death) ~ data[,k] + donor_sex + donor_age_at_diagnosis + project_code,  data= data)
    
    save(fitCox0, file=paste("./Cox_simple/",colnames(data)[k], ".RData"))
    save(fitCoxComplet, file=paste("./Cox_complet/", colnames(data)[k], ".RData"))
    
  }
}

sapply(2:ncol(tab), cox, data=fusion_na)

```

```{r}
cox.zph(fitCox0)
plot(cox.zph(fitCox0))

```

